testplan/web_ui/testing/src/AssertionPane/AssertionTypes/TableAssertions/tableAssertionUtils.js

```diff
@@ -30,22 +61,65 @@ function tableCellStyle(params) {
 
 function tableLogCellRender(cell) {
   if (typeof cell.value === "object") {
-    if (cell.value.type === "link") {
-      return (
-        <a
-          rel="noreferrer"
-          href={cell.value.link}
-          target={cell.value.new_window ? "_blank" : "_self"}
-        >
-          {cell.value.title || cell.value.link}
-        </a>
-      );
+    switch (cell.value.type) {
+      case "link":
+        return <LinkRender {...cell.value}/>;
+      case "formattedValue":
+        return <FormattedValueRender {...cell.value} />;
+      default:
+        break;
+    }
+  }
+  return <>{cell.value}</>;
+}
+
+function tableLogComparator(valueA, valueB, nodeA, nodeB, isInverted) {
+  if (typeof valueA === "object") {
+    switch (valueA.type) {
+      case "link":
+        return tableLogComparator(
+          valueA.title || valueA.link,
+          valueB.title || valueB.link,
+          nodeA,
+          nodeB,
+          isInverted
+        );
+      case "formattedValue":
+        return tableLogComparator(
+          valueA.value,
+          valueB.value,
+          nodeA,
+          nodeB,
+          isInverted
+        );
+      default:
+        return tableLogComparator(
+          parseFloat(valueA),
+          parseFloat(valueB),
+          nodeA,
+          nodeB,
+          isInverted
+        );
     }
   }
-  return cell.value;
+  return valueA >= valueB ? 1 : -1;
 }
 
 
+export function processCellForClipboard(param) {
+  if (typeof param.value === "object") {
+    switch (param.value.type) {
```

Comment:
we have this switch case too many times in this file which means the behavior of the link or formatted type is describe at least 3 places.  As the data is preprocessed before passed to the table, we could create actual classes of these types which implement an interface that is:

- able to return the renderer for itself
- able to return the value for comparison
- return the value for csv export

this way if a new type coming only a single class need to be created that implementing the type and that only need to be hooked up with the parser in a single place.